Tools/autotest/quadplane.py

```diff
@@ -78,6 +78,147 @@ def get_disarm_delay(self):
+        Motor1_value = 33
+        if (self.get_parameter('SERVO5_FUNCTION') != Motor1_value):
+            raise PreconditionFailedException("SERVO5_FUNCTION not %d" % Motor1_value)
+
+        """Make sure that flightmode channel is 5 (default)"""
+        default_fltmode_ch = 5
+        if (self.get_parameter("FLTMODE_CH") != default_fltmode_ch):
+            raise PreconditionFailedException("FLTMODE_CH not %d" % default_fltmode_ch)
+
+        """When disarmed, motor PWM will drop to min_pwm"""
+        #min_pwm = self.get_parameter('SERVO5_MIN')
+        """SERVO5_MIN is not respected; it defaults to 1100, but disarmed PWM is 1000"""
+        min_pwm = self.get_parameter("Q_THR_MIN_PWM")
+
+        """set Q_OPTIONS bit AIRMODE"""
+        airmode_option_bit = (1<<9)
+        self.set_parameter("Q_OPTIONS", airmode_option_bit)
+
+        """set  RC8_OPTION to ARMDISARM"""
+        armdisarm_option = 41
+        arm_ch = 8
+        self.set_parameter("RC%d_OPTION" % arm_ch, armdisarm_option)
+
+        """arm with GCS, record Motor1 SPIN_ARM PWM output and disarm"""
+        spool_delay = self.get_parameter("Q_M_SPOOL_TIME") + 0.25
+        self.zero_throttle()
+        self.arm_vehicle()
+        self.progress("Waiting for Motor1 to spool up to SPIN_ARM")
+        self.delay_sim_time(spool_delay)
+        spin_arm_pwm = self.wait_servo_channel_value(5, min_pwm, comparator=operator.gt)
+        self.progress("spin_arm_pwm: %d" % spin_arm_pwm)
+        self.disarm_vehicle()
+
+        """arm with switch, record Motor1 SPIN_MIN PWM output and disarm"""
+        self.set_rc(8, 2000)
+        self.delay_sim_time(spool_delay)
+        self.progress("Waiting for Motor1 to spool up to SPIN_MIN")
+        spin_min_pwm = self.wait_servo_channel_value(5, spin_arm_pwm, comparator=operator.gt)
+        self.progress("spin_min_pwm: %d" % spin_min_pwm)
+        self.set_rc(8, 1000)
+
+        if (spin_arm_pwm >= spin_min_pwm):
+            raise PreconditionFailedException("SPIN_MIN pwm not greater than SPIN_ARM pwm")
+
+        """verify that arming with switch results in higher PWM output"""
+        self.progress("########## Test auxswitch arming with Q_OPTIONS=AirMode")
+        modes = ('QSTABILIZE', 'QACRO')
+        for mode in modes:
+            self.progress("########## Testing %s mode" % mode)
+            self.change_mode(mode)
+            self.zero_throttle()
+            self.progress("arming with switch at zero throttle")
+            self.arm_motors_with_switch(arm_ch)
+            self.progress("Waiting for Motor1 to speed up")
+            self.wait_servo_channel_value(5, spin_min_pwm, comparator=operator.ge)
+
+            self.progress("disarming with switch")
+            self.disarm_motors_with_switch(arm_ch)
+            self.progress("Waiting for Motor1 to stop")
+            self.wait_servo_channel_value(5, min_pwm, comparator=operator.le)
+            self.wait_ready_to_arm()
```

Comment:
We usually only do this going into a test rather than leaving it.  Not fussed 'though.

Reply:
noted